src/core/net/dnssd_server.hpp

```diff
@@ -300,6 +315,7 @@ class Server : public InstanceLocator, private NonCopyable
         NameCompressInfo mCompressInfo;
         Ip6::MessageInfo mMessageInfo;
         TimeMilli        mStartTime;
+        Server *         mServer;
```

Comment:
`Server` is an `InstanceLocator`. 

We can get `Server` directly if `QueryTransaction` is also a `InstanceLocator`?

Reply:
If we use `InstanceLocator` here, we have to provide an `Instance &` to the constructor of `InstanceLocator`. That means we may need to pass the `Instance &` as an argument to the constructor of `QueryTransaction`. 

However, `Server` owns an array of `QueryTransaction`s . It's tricky to construct such an array of objects who don't have a default constructor.

Reply:
Good point. Thanks.

Reply:
We can consider using `InstanceLocatorInit` which provides an `Init()` function instead (this is what we use for other types which are used in array, e.g. `Neighbor`.